Client/Frontend/Widgets/BlurTableViewController.swift

```diff
@@ -0,0 +1,182 @@
+    var tableView = UITableView()
+    lazy var tapRecognizer: UITapGestureRecognizer = {
+        let tapRecognizer = UITapGestureRecognizer()
+        tapRecognizer.addTarget(self, action: #selector(BlurTableViewController.dismiss(_:)))
+        tapRecognizer.numberOfTapsRequired = 1
+        tapRecognizer.cancelsTouchesInView = false
+        return tapRecognizer
+    }()
+
+    lazy var visualEffectView : UIVisualEffectView = {
+        let visualEffectView = UIVisualEffectView(effect: UIBlurEffect(style: .Light))
+        visualEffectView.frame = self.view.bounds
+        visualEffectView.alpha = 0.90
+        return visualEffectView
+    }()
+
+    lazy var bookmarkAction : BlurTableViewAction = {
+        return BlurTableViewAction(title: NSLocalizedString("Bookmark", comment: "Context Menu Action for Activity Stream"), iconString: "action_bookmark", handler: { action in
+            let shareItem = ShareItem(url: self.site.url, title: self.site.title, favicon: self.site.icon)
+            self.profile.bookmarks.shareItem(shareItem)
+            if #available(iOS 9, *) {
+                var userData = [QuickActions.TabURLKey: shareItem.url]
+                if let title = shareItem.title {
+                    userData[QuickActions.TabTitleKey] = title
+                }
+                QuickActions.sharedInstance.addDynamicApplicationShortcutItemOfType(.OpenLastBookmark,
+                    withUserData: userData,
+                    toApplication: UIApplication.sharedApplication())
+            }
+        })
+    }()
+
+    lazy var deleteFromHistoryAction : BlurTableViewAction = {
+        return BlurTableViewAction(title: NSLocalizedString("Delete from History", comment: "Context Menu Action for Activity Stream"), iconString: "action_delete", handler: { action in
+            self.profile.history.removeHistoryForURL(self.site.url)
+        })
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.actions = [bookmarkAction, deleteFromHistoryAction]
+        view.backgroundColor = UIColor.clearColor().colorWithAlphaComponent(0.4)
+
+        view.addGestureRecognizer(tapRecognizer)
+
+        view.addSubview(tableView)
+        tableView.snp_makeConstraints { make in
+            make.center.equalTo(self.view)
+            make.width.equalTo(290)
+            make.height.equalTo(73 + actions.count * 56)
+        }
+
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.keyboardDismissMode = UIScrollViewKeyboardDismissMode.OnDrag
+        tableView.backgroundColor = UIConstants.PanelBackgroundColor
+        tableView.scrollEnabled = false
+        tableView.separatorColor = UIConstants.SeparatorColor
+        tableView.layer.cornerRadius = 10
+        tableView.registerClass(BlurTableViewCell.self, forCellReuseIdentifier: "BlurTableViewCell")
+        tableView.registerClass(BlurTableViewHeaderCell.self, forCellReuseIdentifier: "BlurTableViewHeaderCell")
```

Comment:
Nitpicky but I would just reorder this code to have the constraints setup last and the table view setup code first so the table view is setup before we add it.